blueprint/core/blueprint_ReactApplicationRoot.h

```diff
@@ -195,35 +189,31 @@ namespace blueprint
             // Clear error state from previous js evals
             errorText.reset();
 
-            if (!bundleValid)
```

Comment:
This bundleValid check was originally here to stop us from calling `registerNativeRenderingHooks` for a second time which results in duplicate function entries in the duktape stack. I think removing this opens us up for that issue again in hot-reload scenarios? 

i.e. you make a valid edit which causes no error so the stack is not cleared. Then we call evaluate in response to the bundle change and re-register our "native rendering hooks"

Reply:
Ahh yes, another great catch. Hm, interesting. On hot reload should we reset the engine? I.e. clear the value stack and reset the release pool?

Reply:
Yeah I think we discussed this briefly during the debugger work. We wan't to keep the Engine and the duk ctx alive to persist debugger connections during reloads (for debugging bundle load etc.). So we could expose some sort of `reset()` or `clear()` function on the engine to do this? Obviously this should all go into `handleBundleChanged` so that we re-register ViewManager and everything else etc.

Reply:
Ugh, of course, that totally slipped my mind. Very glad I have you looking over this right now :D

Yea let's add a `reset()` that keeps the duk_context alive but resets the value stack. I'll update shortly

Reply:
Heh. No worries, glad to be of service on this end for once! Normally you correcting my hacks!